src/modules/drop-erc-721.ts

```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
+        await this.contractWrapper.readContract.getClaimConditionAtIndex(i),
+      );
+    }
+    return Promise.all(
+      conditions.map((c) => this.transformResultToClaimCondition(c)),
+    );
+  }
+
+  /**
+   * Get the total supply for this Drop.
+   *
+   * @returns the total supply
+   */
+  public async totalSupply(): Promise<BigNumber> {
+    return await this.contractWrapper.readContract.nextTokenIdToMint();
+  }
+
+  /**
+   * Get the unclaimed supply for this Drop.
+   *
+   * @returns the unclaimed supply
+   */
+  public async totalUnclaimedSupply(): Promise<BigNumber> {
+    return (await this.contractWrapper.readContract.nextTokenIdToMint()).sub(
+      await this.totalClaimedSupply(),
+    );
+  }
+
+  /**
+   * Get the claimed supply for this Drop.
+   *
+   * @returns the claimed supply
+   */
+  public async totalClaimedSupply(): Promise<BigNumber> {
+    return await this.contractWrapper.readContract.nextTokenIdToClaim();
+  }
+
+  /**
+   * Get NFT Balance
+   *
+   * @remarks Get a wallets NFT balance (number of NFTs in this module owned by the wallet).
+   *
+   * @example
+   * ```javascript
+   * // Address of the wallet to check NFT balance
+   * const address = "{{wallet_address}}";
+   *
+   * const balance = await module.balanceOf(address);
+   * console.log(balance);
+   * ```
+   */
+  public async balanceOf(address: string): Promise<BigNumber> {
+    return await this.contractWrapper.readContract.balanceOf(address);
+  }
+
+  /**
+   * Get NFT Balance for the currently connected wallet
+   */
+  public async balance(): Promise<BigNumber> {
+    return await this.balanceOf(await this.contractWrapper.getSignerAddress());
+  }
+
+  /**
+   * Get whether this wallet has approved transfers from the given operator
+   * @param address - the wallet address
+   * @param operator - the operator address
+   */
+  public async isApproved(address: string, operator: string): Promise<boolean> {
+    return await this.contractWrapper.readContract.isApprovedForAll(
+      address,
+      operator,
+    );
+  }
+
+  /**
+   * Get the primary sale recipient.
+   * @returns the wallet address.
+   */
+  public async getPrimarySaleRecipient(): Promise<string> {
+    return await this.contractWrapper.readContract.primarySaleRecipient();
+  }
+
+  /** ******************************
+   * PRIVATE FUNCTIONS
+   *******************************/
+
+  private async getTokenMetadata(tokenId: string): Promise<NFTMetadata> {
+    // return await getTokenMetadata(
+    //   this.contractWrapper.readContract,
+    //   tokenId,
+    //   this.storage.ipfsGatewayUrl,
+    // );
+    // TODO common token metadata fetch
+    return Promise.resolve({});
+  }
+
+  private async transformResultToClaimCondition(
+    pm: IDropERC721.ClaimConditionStructOutput,
+  ): Promise<ClaimCondition> {
+    // TODO have a dedicated class for currency manipulation that takes in a contractWrapper?
+    // const cv = await getCurrencyValue(
```

Comment:
same for this one - feels like all the currency stuff could also be encapsulated in a class